content/4_Gold/Divis.mdx

```diff
@@ -448,19 +448,10 @@ def gcd(a, b):
 
 </LanguageSection>
 
-This function runs in $\mathcal{O}(\log ab)$ time because
-$a\le b \implies b\pmod a <\frac{b}{2}$.
-
-The worst-case scenario for the Euclidean algorithm is when $a$ and $b$ are
-consecutive Fibonacci numbers $F_n$ and $F_{n + 1}$. for an explanation). In
-this case, the algorithm will calculate
-$\gcd(F_n, F_{n + 1}) = \gcd(F_{n - 1}, F_n) = \dots = \gcd(0, F_1)$. This means
-that finding $\gcd(F_n, F_{n + 1})$ takes $n + 1$ steps, which is proportional
-to $\log \left(F_n F_{n+1}\right)$.
+This function runs in $\mathcal{O}(\log a + \log b)$ time because
+$a\le b \implies b \mod a < \frac{a + (b \mod a)}{2} < \frac{b}{2}$.
 
-<!-- (See [this Math Stack Exchange answer](https://math.stackexchange.com/questions/2477328/why-are-fibonacci-numbers-bad-for-euclids-algorithm-and-how-to-derive-this-uppe) -->
-<!-- ^ this doesn't seem very helpful -->
-<!-- Since $F_n = O(\varphi^n)$, the worst-case complexity of this algorithm is $\mathcal{O}(\log ab)$. -->
+The worst-case scenario for the Euclidean algorithm is when $a$ and $b$ are consecutive Fibonacci numbers $F_n$ and $F_{n + 1}$. In this case, the algorithm will calculate $\gcd(F_n, F_{n + 1}) = \gcd(F_{n - 1}, F_n) = \dots = \gcd(0, F_1)$. This means that finding $\gcd(F_n, F_{n + 1})$ takes $n + 1$ steps, which is proportional to $\log F_n + \log F_{n+1}$.
```

Comment:
What's wrong with the original explanation?

Reply:
There are small formatting mistakes and I think a few parts of the complexity analysis are unclear.